server-channel.go

```diff
@@ -0,0 +1,128 @@
+package main
+
+import (
+	"fmt"
+	"os/exec"
+	"strconv"
+	"strings"
+)
+
+type ServerChannel interface {
```

Comment:
I don't like the name either - it's actually an abstraction onto the Machine we are deploying to - so DeployBox or DeployMachine?